### PR TITLE
Switch from lerna to pnpm

### PR DESCRIPTION
**Pull request by evantahler**
`pnpm` is faster, shares dependencies and can still support monorepos.

Tasks:

- [x] Install PNPM
- [x] Change install scripts
- [x] Change CI scripts
- [x] Enforce PNPM usage
- [x] Revisit CI Cache
- [x] Tests Pass
- [x] Sort out Publishing (we can continue to use Lerna)

Notes:

* We can still use Lerna to publish and update versons... just not for managing dependencies. 

Closes T-655

### CODE REVIEW

**Comment by evantahler** [comment]
Currently blocked when running tests against a next.js error:

```
    ReferenceError: URL is not defined

      at Object.<anonymous> (../../node_modules/.pnpm/next@9.5.6-canary.13_8276e7ef4e4bcb622af0fe7b5225f62d/node_modules/next/next-server/lib/router/utils/parse-relative-url.ts:8:1)
      at Object.<anonymous> (../../node_modules/.pnpm/next@9.5.6-canary.13_8276e7ef4e4bcb622af0fe7b5225f62d/node_modules/next/next-server/lib/router/utils/prepare-destination.ts:4:1)
      at Object.<anonymous> (../../node_modules/.pnpm/next@9.5.6-canary.13_8276e7ef4e4bcb622af0fe7b5225f62d/node_modules/next/next-server/server/next-server.ts:65:1)
      at Object.<anonymous> (../../node_modules/.pnpm/next@9.5.6-canary.13_8276e7ef4e4bcb622af0fe7b5225f62d/node_modules/next/server/next.ts:3:1)
      at Object.<anonymous> (../../node_modules/.pnpm/ah-next-plugin@0.3.0_5139af8b50ead0a453273331284dbaed/node_modules/ah-next-plugin/dist/initializers/next.js:8:32)
```

**Comment by evantahler** [comment]
The Next error was resolved by https://github.com/actionhero/ah-next-plugin/pull/115

**Comment by evantahler** [comment]
@bleonard can you give this branch a whirl to make sure everything works for you?  start with `npm run nuke` and then try `pnpm install`.  It should install all the deps and build all the packages... but faster!  Also, check out both `npm run start` and `npm run dev`... which should preform about the same as they do on `master`.

**Comment by evantahler** [comment]
If this PR is merged - Update https://www.grouparoo.com/docs/development

**Comment by bleonard** [comment]
Started with fresh directory
`> npm i -g pnpm`
`> pnpm install`

`apps/staging-public> npm run dev` seems to work.

`apps/staging-public> npm start` has this error:

```
➜  staging-public git:(pnpm) npm start

> @grouparoo/app-staging-public@0.1.23 start /Users/brian/grouparoo/test/grouparoo/apps/staging-public
> cd node_modules/@grouparoo/core && GROUPAROO_MONOREPO_APP=staging-public ./api/bin/start

modified your runtime environment with /Users/brian/grouparoo/test/grouparoo/apps/staging-public/.env

!!! plugin @grouparoo/files-s3 requires environment variable S3_ACCESS_KEY to be set !!!


!!! plugin @grouparoo/files-s3 requires environment variable S3_SECRET_ACCESS_KEY to be set !!!


!!! plugin @grouparoo/files-s3 requires environment variable S3_REGION to be set !!!


!!! plugin @grouparoo/files-s3 requires environment variable S3_BUCKET to be set !!!


!!! plugin @grouparoo/newrelic requires environment variable NEW_RELIC_LICENSE_KEY to be set !!!


!!! plugin @grouparoo/newrelic requires environment variable NEW_RELIC_APP_NAME to be set !!!

not injecting newrelic because NEW_RELIC_LICENSE_KEY is not set
2020-10-29T23:55:41.902Z - notice: Starting Grouparoo 0.1.23 on node.js v12.16.3 
2020-10-29T23:55:42.395Z - info: UNHANDLED REJECTION: Error: Cannot find module 'validator/lib/isURL'
Require stack:
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/modules/ops/profileProperty.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/ProfileProperty.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/ProfilePropertyRule.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/GroupRule.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/Group.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/GroupMember.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/Profile.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/Run.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/Schedule.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/Source.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/models/App.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/modules/plugin.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/index.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/initializers/events.js
- /Users/brian/grouparoo/test/grouparoo/node_modules/.pnpm/actionhero@24.0.1/node_modules/actionhero/dist/classes/process.js
- /Users/brian/grouparoo/test/grouparoo/node_modules/.pnpm/actionhero@24.0.1/node_modules/actionhero/dist/index.js
- /Users/brian/grouparoo/test/grouparoo/core/api/dist/grouparoo.js

```

**Comment by bleonard** [comment]
Stats from my computer.
master: `npm install  1028.81s user 236.01s system 138% cpu 15:16.08 total`
pnpm: `pnpm install  537.00s user 53.32s system 166% cpu 5:54.92 total`

3x faster 🎉

**Comment by bleonard** [comment]
`npm start` is looking good now.

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+packages:
+  - "core"
+  - "cli"
+  - "clients/**"
+  - "plugins/**"
+  - "apps/**"
+  - "!apps/local-public"
```
Comment (bleonard) [inline pnpm-workspace.yaml:7]: why not this one?

Comment (evantahler) [reply pnpm-workspace.yaml:7]: We ignore it in Lerna-land as well https://github.com/grouparoo/grouparoo/blob/master/lerna.json#L11-L13

Just so we don't need to wait for 2 apps to build when installing

**Pull request merged**